compiler/src/dmd/cppmanglewin.d

```diff
@@ -1301,3 +1124,188 @@ extern(D):
         return ret;
     }
 }
+
+private:
+extern(D):
+
+/**
+ * Computes mangling for symbols with special mangling.
+ * Params:
+ *      sym = symbol to mangle
+ * Returns:
+ *      mangling for special symbols,
+ *      null if not a special symbol
+ */
+string mangleSpecialName(Dsymbol sym)
+{
+    string mangle;
+    if (sym.isCtorDeclaration())
+	mangle = "?0";
+    else if (sym.isAggregateDtor())
+	mangle = "?1";
+    else if (!sym.ident)
+	return null;
+    else if (sym.ident == Id.assign)
+	mangle = "?4";
+    else if (sym.ident == Id.eq)
+	mangle = "?8";
+    else if (sym.ident == Id.index)
+	mangle = "?A";
+    else if (sym.ident == Id.call)
+	mangle = "?R";
+    else if (sym.ident == Id.cppdtor)
+	mangle = "?_G";
+    else
+	return null;
+
+    return mangle;
+}
+
+/**
+ * Mangles an operator, if any
+ *
+ * Params:
+ *      buf                 = buffer to write mangling to
+ *      ti                  = associated template instance of the operator
+ *      symName             = symbol name
+ *      firstTemplateArg    = index if the first argument of the template (because the corresponding c++ operator is not a template)
+ * Returns:
+ *      true if sym has no further mangling needed
+ *      false otherwise
+ */
+bool mangleOperator(ref OutBuffer buf, TemplateInstance ti, ref const(char)[] symName, ref int firstTemplateArg)
+{
+    auto whichOp = isCppOperator(ti.name);
+    final switch (whichOp)
+    {
+    case CppOperator.Unknown:
+	return false;
+    case CppOperator.Cast:
+	buf.writestring("?B");
+	return true;
+    case CppOperator.Assign:
+	symName = "?4";
+	return false;
+    case CppOperator.Eq:
+	symName = "?8";
+	return false;
+    case CppOperator.Index:
```

Comment:
indentation is all off

Reply:
Yes, it looks correct when tabs are 8 columns, but when you configure GitHub to make them 4 columns, you'll see the difference. In any case, the tabs should be converted to spaces to be consistent with the rest here.

Reply:
Looks like I forgot to detab them. I thought the style checker would complain about that. Oh well. Detabbed.